src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,447 @@
+
+        /** Avatar url of participant */
+        avatar: PropTypes.string,
+    }),
+
+    /** Personal details of the current user */
+    currentUserPersonalDetails: PropTypes.shape({
+        // Local Currency Code of the current user
+        localCurrencyCode: PropTypes.string,
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    hasMultipleParticipants: false,
+    report: {
+        participants: [],
+    },
+    iouType: CONST.IOU.IOU_TYPE.REQUEST,
+    currentUserPersonalDetails: {
+        localCurrencyCode: CONST.CURRENCY.USD,
+    },
+    personalDetails: {},
+    iou: {
+        creatingIOUTransaction: false,
+        error: false,
+        selectedCurrencyCode: null,
+    },
+};
+
+// Determines type of step to display within Modal, value provides the title for that page.
+const Steps = {
+    IOUAmount: 'iou.amount',
+    IOUParticipants: 'iou.participants',
+    IOUConfirm: 'iou.confirm',
+};
+
+const MoneyRequestModal = (props) => {
+    const reportParticipants = lodashGet(props, 'report.participants', []);
+    const participantsWithDetails = _.map(OptionsListUtils.getPersonalDetailsForLogins(reportParticipants, props.personalDetails), personalDetails => ({
+        login: personalDetails.login,
+        text: personalDetails.displayName,
+        firstName: lodashGet(personalDetails, 'firstName', ''),
+        lastName: lodashGet(personalDetails, 'lastName', ''),
+        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
+        icons: [{
+            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
+            name: personalDetails.login,
+            type: CONST.ICON_TYPE_AVATAR,
+        }],
+        keyForList: personalDetails.login,
+        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
+        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
+    }));
+
+    // Skip IOUParticipants step if participants are passed in
+    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];
+
+    const prevCreatingIOUTransactionStatusRef = useRef(lodashGet(props.iou, 'creatingIOUTransaction'));
+    const prevNetworkStatusRef = useRef(props.network.isOffline);
+
+    const [previousStepIndex, setPreviousStepIndex] = useState(0);
+    const [currentStepIndex, setCurrentStepIndex] = useState(0);
+    const [participants, setParticipants] = useState(participantsWithDetails);
+    const [amount, setAmount] = useState('');
+    const [comment, setComment] = useState('');
+
+    useEffect(() => {
+        PersonalDetails.openMoneyRequestModalPage();
+        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- props.currentUserPersonalDetails will always exist from Onyx
```

Comment:
Just out of curiousity, what is the linter complaining about here?

`react-hooks/exhaustive-deps`

Reply:
We just want to run this on mount, and because I'm using `props.currentUserPersonalDetails.localCurrencyCode` within the effect it wants me to add it as a dependency - but it's not needed because we get it from onyx

Reply:
>it's not needed because we get it from onyx

And because we do not want this effect to run again.

I think it's kind of an anti-pattern though (that we can clean up later probably). If we always want the "iou selected currency" to be the "local currency code" when this component mounts then we should update that value in Onyx whenever we update the local currency code. We can hook into the initial render like this - but it's not obvious if we should...